Added AVL Tree Visualizer project

anirudh-248 opened the pull request:
# Pull Request for PyVerse 💡


## Issue Title : <!--  -->AVL Tree Visualizer #197

- **Info about the related issue (Aim of the project)** : <!-- What's the goal of the project --> The aim of the project is to create a visualization tool that helps users visualize the construction of an AVL Tree and get to know how it works. It is also used to sort the given input elements.
- **Name:** <!-- -->Anirudh P S
- **GitHub ID:** <!--  -->anirudh-248
- **Email ID:** <!--  -->anirudhphaniraj@gmail.com
- **Identify yourself: (Mention in which program you are contributing in. Eg. For a WoB 2024 participant it's, `WoB Participant`)** <!--  -->GSSoC Contributor / Hacktoberfest


<!-- Mention the following details and these are mandatory -->

Closes: #197 

### Describe the add-ons or changes you've made 📃

This interactive web app, built with Streamlit, allows users to visualize AVL tree construction in real time. Users can upload files or manually input numbers to observe step-by-step updates of the tree structure. Features include traversal results, immediate feedback on user inputs, and detailed visualizations, making it an ideal tool for learning and teaching AVL trees interactively.

## Type of change ☑️

What sort of change have you made:
<!--
Example of how to mark a checkbox:-
- [x] My code follows the code style of this project.
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

The project has been tested by providing multiple input files and varying sequences of input numbers. The AVL Tree is built correctly each time by performing the correct rotations required. It also generates an output file that contains the elements of the input in a sorted manner.

## Checklist: ☑️
<!--
Example of how to mark a checkbox:-
- [x] My code follows the code style of this project.
-->
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

@UTSAVS26 Please review this PR and let me know if any changes are required.

UTSAVS26 reviewed (changes requested):
Please add readme file of this [format](https://github.com/UTSAVS26/PyVerse/blob/main/.github%2Freadme_template.md).
And add comments in the code

anirudh-248 commented:
@UTSAVS26 I have updated the readme file to the requested format and I have also added comments in my code. Please let me know if any further changes are required.
